src/ansi-c/expr2c.cpp

```diff
@@ -421,33 +424,50 @@ std::string expr2ct::convert_rec(
   else if(src.id()==ID_c_enum_tag)
   {
     const c_enum_tag_typet &c_enum_tag_type=to_c_enum_tag_type(src);
-    const symbolt &symbol=ns.lookup(c_enum_tag_type);
     std::string result=q+"enum";
-    result+=" "+id2string(symbol.base_name);
+    if(ns)
+    {
+      const symbolt &symbol = ns->get().lookup(c_enum_tag_type);
+      result += " " + id2string(symbol.base_name);
+    }
+    else
+      result += " " + id2string(c_enum_tag_type.get(ID_identifier));
     result+=d;
     return result;
   }
   else if(src.id()==ID_pointer)
   {
     c_qualifierst sub_qualifiers;
     sub_qualifiers.read(src.subtype());
-    const typet &subtype_followed=ns.follow(src.subtype());
```

Comment:
This `ns.follow` is actually unnecessary as none of the cases considered below would result in a symbol type. (This was different some time in the past, but now is restricted to structs.)

Reply:
Not completely sure, but is that true even with C typedefs?